src/k_lock_mutex.rs

```diff
@@ -210,26 +214,26 @@ impl<T: ?Sized> Mutex<T> {
             let mut spin = 10;
             let state = loop {
                 let v = self.futex.load(Ordering::Relaxed);
-                if v != 1 || spin == 0 {
+                if v != LOCKED || spin == 0 {
                     break v;
                 }
                 spin_loop();
                 spin -= 1;
             };
 
-            if (state == 0
+            if (state == UNLOCKED
                 && self
                     .futex
-                    .compare_exchange(0, 1, Ordering::Acquire, Ordering::Relaxed)
+                    .compare_exchange(UNLOCKED, LOCKED, Ordering::Acquire, Ordering::Relaxed)
                     .is_ok())
-                || (state != 2 && self.futex.swap(2, Ordering::Acquire) == 0)
+                || (state != CONTENDED && self.futex.swap(CONTENDED, Ordering::Acquire) == 0)
```

Comment:
```suggestion
                || (state != CONTENDED && self.futex.swap(CONTENDED, Ordering::Acquire) == UNLOCKED)
```
The 0 in this case has significant meaning